transformer_engine/pytorch/module.py

```diff
@@ -135,6 +135,44 @@ def _prepare_backward(fp8: bool,
         delete_key_from_amax_buffer(forward=False)
 
 
+class NoopCat(torch.autograd.Function):
+    """This class is a no-op replacement for `torch.cat`."""
+
+    @staticmethod
+    def forward(ctx, full_param_buffer, *params_split):
+        assert not full_param_buffer.requires_grad, "Buffers should not require gradient"
+        assert (
+            full_param_buffer.shape[0] % len(params_split) == 0
+        ), "Dimensions not compatible for concatenation"
+
+        split_size = full_param_buffer.shape[0] // len(params_split)
+        for i, p in enumerate(params_split):
+            p.data = full_param_buffer[i * split_size : (i+1) * split_size]
```

Comment:
I don't think I follow the logic here - shouldn't that be a check instead?

As in, I give the set of buffers to this op to be concatenated (and the expectation is that they already are concatenated in memory), so why am I setting their pointers to something potentially different?